main.py

```diff
@@ -356,6 +362,53 @@ def __init__(self, *args):
 		self.actionExit.triggered.connect(self.onExitClicked)
 		self.actionSettings.triggered.connect(self.onSettingsClicked)
 
+		if not "search_for_updates" in self.settings.settings or not "buildId" in self.settings.settings:
```

Comment:
I think most of the following update procedure should be moved into a function (or possibly more than one).